iceoryx_utils/include/iceoryx_utils/design_pattern/creation.hpp

```diff
@@ -23,6 +23,77 @@
 
 namespace DesignPattern
 {
+/// @brief This pattern can be used if you write an abstraction where you have
+///        to throw an exception in the constructor when you for instance would
```

Comment:
Nitpick: We are never throwing exceptions in iceoryx. What about "..you have to signal an error in the constructor.."?